### ISSUE TITLE
About Pen and Brush

### ISSUE THREAD

**Issue by 4ever911** [pyqtgraph/pyqtgraph#388]
Hi, this may not be a issue.

I am a former Windows developer. Since Pen and Brush are kernel objects, to save the resource, we used to create Pens and Brushes outside for loops

for instance:

pen = createpen(...)
for .....loop
       select pen. 
       draw....

and sometimes, we save those Pen/Brushes as global variables.

so, for Python and pyqtgraph, does it really matter?

**Comment by NilsNemitz** [comment]
For the record:
Creating pens and brushes has a certain overhead in internal function calls. Where speed is critical, it might be helpful to pre-generate QPen and QBrush objects, and then pass these directly to the plotting functions. This will generally bypass the generation code.

There are also objects like ScatterPlotItem that can identify previously used QBrush objects and use this to speed up drawing through an internal cache.

In summary, there are good reasons to pre-generate pen and brush objects and re-use them as needed.
But in practice, it is often more convenient to just describe pens and brushes in the calls to the plotting function, and this is generally efficient enough.
